### PR TITLE
[jit] Enable float32 by default.

### PR DESCRIPTION
**Pull request by vargaz**
Implements #6985

### CODE REVIEW

**Comment by vargaz** [comment]
build

**Comment by vargaz** [comment]
@monojenkins build

**Comment by luhenry** [comment]
IIUC, we do need the x86 support for that feature before being able to make the switch on the x64, arm and arm64. @vargaz how long would you expect it to take?

**Comment by luhenry** [comment]
After discussing the matter with @kumpera, we reached the conclusion that the slight behavior difference between Android x86 and Android arm/arm64/x64 is ok, as it's already the case between iOS x86 and iOS arm/arm64/x64.

**Comment by luhenry** [comment]
@vargaz there are a lot of failures in the xunit tests and acceptance tests. They all seems to be related to the change in behavior of floating operations. Is it something that is expected, or a bug in the code?

**Comment by vargaz** [comment]
There are a lot of different failures, so fixing them is going to take some time, not sure we can merge this right now.

**Comment by vargaz** [comment]
A lot of tests depend on auto conversion between r8 and r4, i.e.:
ldc.r4 1.2
box r8
etc.

Without float32, this worked because we auto converted the r4 value to an r8 before pushing it on the il stack, but in float32 mode, we push an r4, leading to invalid IL errors.
The reason these tests don't fail on mobile is that we don't run them.

**Comment by vargaz** [comment]
Will move parts of this to another PR to make it smaller.

**Comment by marek-safar** [comment]
The SLE tests are failing on one of the following values

            float?[] values = new float?[] { null, 0, 1, -1, float.MinValue, float.MaxValue, float.Epsilon, float.NegativeInfinity, float.PositiveInfinity, float.NaN };

which should work as expected so it looks like real issue

**Comment by vargaz** [comment]
This PR is only used to run ci with -O=float32 enabled, the fixes will be in separate prs.

**Comment by vargaz** [comment]
Looks like the remaining failures are:
in coreclr tests: external/coreclr/tests/src/JIT/jit64/opt/regress/vswhidbey/481244/foo2_il.exe
This one is caused by an r4-r8 stack merge and precision loss. We should disable this for now.
on arm64: multiple test failures

File: acceptance-tests/coreclr.mk [hunk 1]
```diff
@@ -5061,6 +5060,9 @@ CORECLR_DISABLED_TEST_IL_SRC +=	\
 CORECLR_DISABLED_TEST_IL_SRC +=	\
 	$(CORECLR_PATH)/tests/src/JIT/Directed/coverage/oldtests/lcliimpl.il
 
+# We use an r4 for r4-r8 stack merge so it loses precision
```
Comment (jaykrell) [inline acceptance-tests/coreclr.mk]: Mixing r4 and r8 leads to r4? But it should lead to r8?

**Pull request merged**